DataContent/DAL/Repositories/UserRepository.cs

```diff
@@ -96,9 +97,19 @@ public async Task<User> UpdateUserAsync(User user)
                 updatedUser.Email = user.Email;
                 updatedUser.Username = user.Username;
                 updatedUser.Password = new ScryptEncoder().Encode(user.Password);
+                updatedUser.RecoveryPassword = null;
             }
             await _context.SaveChangesAsync();
             return updatedUser;
         }
+
+        public async Task<User> ForgotPasswordAsync(string email)
+        {
+            var user = await _context.Users.Where(x => x.Email == email).FirstOrDefaultAsync();
```

Comment:
I think there we need to check, if there was at least 30s or more between recoveries before adding new recovery password

Reply:
Done